File: ABC/B/257/B.rb
n, k, q = gets.chomp.split.map(&:to_i)
hash = Hash.new(0)
gets.chomp.split.map(&:to_i).each_with_index do |val, index| 
    hash[index + 1] = val
end
gets.chomp.split.map(&:to_i).each do |l|
    if hash[l] != n then
        next_to = false
        hash.each do |key, val|
            next_to = true if val == hash[l] + 1
        end
        hash[l] += 1 if !next_to
    end
end
cnt = 1
hash.each do |key, val|
    if cnt == k then p val
    else print "#{val} "
    end
    cnt += 1
end

File: ABC/B/257/B.py
from operator import ne


n, k, q = map(int, input().split())
a = list(map(int, input().split()))
dict = {}
for i in range(k): dict[i+1] = a[i]
for l in list(map(int, input().split())):
    if dict[l] != n:
        next_to = False
        for key in dict:
            if dict[l]+1 == dict[key]: next_to = True
        if not next_to: dict[l] += 1
cnt = 1
for key in dict:
    if cnt != k: print(dict[key], end=' ')
    else: print(dict[key])
    cnt += 1